bindgen/templates/ObjectRuntime.go

```diff
@@ -46,6 +46,11 @@ func (ffiObject *FfiObject)decrementPointer() {
 }
 
 func (ffiObject *FfiObject)destroy() {
+	if ffiObject.pointer == nil {
```

Comment:
I don't see how `FfiObject.pointer` could ever be nil. The struct is supposed to be constructed with a non-nil pointer, and freeing the pointer simply sets `destroyed` flag without modifying `pointer` value.

Reply:
it very much was unfortunately, feel free to comment it out and see the beautiful panic :/

Reply:
my suspicion is that it happens because it is destroyed on the rust side but I never fully figured it out

Reply:
I took a closer look at this. This happens in `TestFuturesFallibleThrows`, more specifically when calling `FallibleStruct(true)`. The panic you are talking about in tests actually happens in another test that triggers the GC. The root cause of this is faulty error handling code in async code. Async code lifts the object when an error has occured in Rust call. This results is nil pointer being used to create Go object, and later the object finalizer attempts to free a nil pointer in Rust.

Adding `if ffiObject.pointer == nil` here is not fixing the actual issue. The real issue is that async code attempts to lift the value when Rust has returned an error. This works fine for some types, but doesn't work for other types. Lifting the default return value is fine for numeric types. It also works for strings, because lifting a string from an empty RustBuffer yields an empty string. Lifting objects is not ok like you already encountered, albeit it can be worked around with this if statement. Lifting some other types, like enums, structs, sequences and maps is even more problematic. Those types expect a RustBuffer that contains the right data, and an empty RustBuffer would cause panic when lifting.

Reply:
thank you for figuring this out, pushed a fix for this